src/index.js

```diff
@@ -79,6 +79,19 @@ class Radar {
     Cookie.setCookie(Cookie.DESCRIPTION, description);
   }
 
+  static setMetadata(metadata) {
+    if (!metadata) {
+      Cookie.deleteCookie(Cookie.METADATA);
+      return;
+    }
+
+    metadata = JSON.stringify(metadata).trim();
+    if (metadata.length === 0 || metadata.length > 256) {
```

Comment:
Why delete if larger than 256?

Reply:
ah, I see - that's the cap on `description` in upsert, `metadata` is capped at 16 keys, changing

Reply:
We also don't need to be doing any validations client-side. It should fail on the server (no need to duplicate the logic here)

Reply:
also realized it's impossible for `JSON.stringify(metadata)` to have length 0 at this branch with the `if (!metadata)` check above, even stringifying `''` yields a nonzero length
```
> JSON.stringify('');
< """"
```